src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -300,17 +308,40 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             // delete IEquatable if it's explicit because it is implicit on records
             var iEquatable = ConvertToRecordHelpers.GetIEquatableType(semanticModel.Compilation, type);
             var baseList = typeDeclaration.BaseList;
-            if (baseList != null && iEquatable != null)
+            if (baseList != null)
             {
-                var typeList = baseList.Types.Where(baseItem
-                    => !iEquatable.Equals(semanticModel.GetTypeInfo(baseItem.Type, cancellationToken).Type));
+                var typeList = baseList.Types.AsImmutable();
```

Comment:
why do we need this to be an array form.  i would think we woudl want to keep this in SeparatedSyntaxList form.